src/ducks/modules/stage.js

```diff
@@ -1,13 +1,18 @@
+import { actionTypes as protocolActionTypes } from '../../ducks/modules/protocol';
+
 const NEXT_PROMPT = 'NEXT_PROMPT';
 const PREVIOUS_PROMPT = 'PREVIOUS_PROMPT';
+const NEXT_STAGE = 'NEXT_STAGE';
+const PREVIOUS_STAGE = 'PREVIOUS_STAGE';
 
 const initialState = {
   promptIndex: 0,
```

Comment:
Okay, so based on an attempt at refactoring that, we need to track prompt states independently. I'll rename the module to `participantProgress` and add the additional state data, so you can take a look.